src/main/java/javax/jmdns/impl/HostInfo.java

```diff
@@ -104,6 +106,11 @@ public static HostInfo newHostInfo(InetAddress address, JmDNSImpl dns, String jm
         if (index > 0) {
             aName = aName.substring(0, index);
         }
+        if (aName.length() > _labelLengthLimit) {
+            // Remove trailing labels which would make the combined label exceed 63 characters in length
```

Comment:
Could you maybe add an example in the comment on what a too-long-label would be shortened to? (i.e. show why to better cut it before the last "." instead of simply limiting it to 63 chars.

Reply:
It could have been done in other ways, I agree. It was just what seemed most natural to me.
Following the example above, the resulting name would then be fv-az767-139-5kzmkpjakaaulcrpfkoga5aama-gx-internal-cloudapp.local

Reply:
The DNS spec ([RFC 1035 - Section 2.3.1](https://www.ietf.org/rfc/rfc1035.html#section-2.3.1)) is based on plain text. In this case every character would be an octet.
mDNS however is based on UTF-8 (see [Multicast DNS Character Set](https://datatracker.ietf.org/doc/html/rfc6762#section-16)).
As the maximum number of octets per label has not changed one would need to check the number of bytes needed to encode a given UTF-8 string against the limit of 63 octets.
Cutting off based on the limit might be also more complex because "simply" cutting of after the limit could cut through an encoded UTF-8 character making the resulting label invalid UTF-8.